arangod/Replication2/StateMachines/Document/DocumentFollowerState.cpp

```diff
@@ -115,17 +115,13 @@ auto DocumentFollowerState::applyEntries(
 
             if (doc.operation == OperationType::kAbortAllOngoingTrx) {
               self->_activeTransactions.clear();
-              if (!releaseIndex.has_value() ||
-                  releaseIndex.value() < entry->first) {
-                releaseIndex = entry->first;
-              }
+              releaseIndex =
+                  self->_activeTransactions.getReleaseIndex(entry->first);
             } else if (doc.operation == OperationType::kCommit ||
                        doc.operation == OperationType::kAbort) {
               self->_activeTransactions.erase(doc.tid);
-              if (!releaseIndex.has_value() ||
-                  releaseIndex.value() < entry->first) {
-                releaseIndex = entry->first;
-              }
+              releaseIndex =
+                  self->_activeTransactions.getReleaseIndex(entry->first);
```

Comment:
the old code computed the maximum of some values. I assume that `getReleaseIndex` returns a monotonically increasing index?

Reply:
Yes, it returns a monotonically increasing index. The difference is that `getReleaseIndex` makes sure that the returned index does not split an active transaction in two halves. In other words, it returns the greatest index so far, that is less than the first occurrence index of any *active* transaction.